### ISSUE TITLE
Displaying room settings for a room of which I'm admin displays 'Unnamed room' and 'Add topic'

### ISSUE THREAD

**Issue by matrixbot** [vector-im/vector-web#1226]
Created by @ Amandine:matrix.org.

The problem doesn't exist if I'm not admin

**Comment by ara4n** [comment]
i can't repro this

**Comment by AmandineLP** [comment]
100% reproduceable, even after refreshing Vector
Tested with #Marketing:matrix.org (!ZzndDZQJtLEsLiMTbY:matrix.org) changing name from "#Marketing:matrix.org," to "gg" to "#Marketing". After setting a new name it appears in the settings (original name; I display settings ->" Unnamed" is displayed; I set a new name "gg"; save; get the "M_UNKNOWN: 'topic' not in content" usual error; display settings again -< topic is still empty but now I see the name as "gg")
```
RoomSettings.js:96 Original: {"topic":"Inception... - Events tracker: https://docs.google.com/spreadsheets/d/19c5B_khrskB_z44ETrFAh1qn0lfg0wJOEKBYohKQEEs/edit?usp=sharing","join_rule":"invite","power_levels_changed":false,"tags_changed":false,"tags":{"m.favourite":{}},"areNotifsMuted":false}
2016-03-20 08:50:39.269 RoomSettings.js:97 New: {"name":"gg","join_rule":"invite","power_levels_changed":false,"tags_changed":false,"tags":{"m.favourite":{}},"areNotifsMuted":false,"_loading":false}
2016-03-20 08:50:39.271 RoomSettings.js:168 Performing 2 operations
2016-03-20 08:50:39.387 matrix.org/_matrix/client/r0/rooms/!ZzndDZQJtLEsLiMTbY%3Amatrix.org/state/m…AyMTM5NAowMDJmc2lnbmF0dXJlIINdx3xXN7zbvQRgdf1pHywhb13QckFlIlIfbwZSzQfnCg:1 PUT https://matrix.org/_matrix/client/r0/rooms/!ZzndDZQJtLEsLiMTbY%3Amatrix.org…ODAyMTM5NAowMDJmc2lnbmF0dXJlIINdx3xXN7zbvQRgdf1pHywhb13QckFlIlIfbwZSzQfnCg 400 (Bad Request)
2016-03-20 08:50:39.424 RoomView.js:824 Settings saved with 1 errors
2016-03-20 08:50:39.424 RoomView.js:827 e.e…s.MatrixError {errcode: "M_UNKNOWN", name: "M_UNKNOWN", message: "'topic' not in content", data: Object, httpStatus: 400}(anonymous function) @ RoomView.js:827(anonymous function) @ RoomView.js:826r @ q.js:834s.promiseDispatch.c @ q.js:863r.promiseDispatch @ q.js:796(anonymous function) @ q.js:604r @ q.js:137e @ q.js:125(anonymous function) @ main.js:64i.run @ browser.js:64r @ browser.js:34
2016-03-20 08:50:39.470 ScrollPanel.js:437 Scrolled to bottom; offset now 2188
2016-03-20 08:50:39.471 ScrollPanel.js:254 ScrollPanel: starting f fill
2016-03-20 08:50:39.731 ScrollPanel.js:437 Scrolled to bottom; offset now 2188
2016-03-20 08:50:39.731 ScrollPanel.js:250 ScrollPanel: Already a f fill in progress - not starting another
2016-03-20 08:50:39.789 ScrollPanel.js:437 Scrolled to bottom; offset now 2297
2016-03-20 08:50:39.789 ScrollPanel.js:254 ScrollPanel: starting f fill
2016-03-20 08:50:39.790 ScrollPanel.js:272 ScrollPanel: f fill complete; hasMoreResults:false
2016-03-20 08:50:39.790 ScrollPanel.js:272 ScrollPanel: f fill complete; hasMoreResults:false
2016-03-20 08:50:39.791 ScrollPanel.js:142 Scroll event: offset now: 2297 recentEventScroll: undefined
2016-03-20 08:50:39.791 ScrollPanel.js:172 Ignoring scroll echo
2016-03-20 08:50:39.792 ScrollPanel.js:254 ScrollPanel: starting f fill
2016-03-20 08:50:39.807 ScrollPanel.js:272 ScrollPanel: f fill complete; hasMoreResults:false
2016-03-20 08:50:39.859 ScrollPanel.js:437 Scrolled to bottom; offset now 2297
2016-03-20 08:50:39.860 ScrollPanel.js:254 ScrollPanel: starting f fill
2016-03-20 08:50:39.871 ScrollPanel.js:272 ScrollPanel: f fill complete; hasMoreResults:false
2016-03-20 08:51:05.144 ScrollPanel.js:437 Scrolled to bottom; offset now 2703
2016-03-20 08:51:05.145 ScrollPanel.js:254 ScrollPanel: starting f fill
2016-03-20 08:51:05.155 ScrollPanel.js:272 ScrollPanel: f fill complete; hasMoreResults:false
2016-03-20 08:51:05.169 ScrollPanel.js:142 Scroll event: offset now: 2703 recentEventScroll: undefined
2016-03-20 08:51:05.170 ScrollPanel.js:172 Ignoring scroll echo
2016-03-20 08:51:05.171 ScrollPanel.js:254 ScrollPanel: starting f fill
2016-03-20 08:51:05.179 ScrollPanel.js:272 ScrollPanel: f fill complete; hasMoreResults:false
2016-03-20 08:51:30.275 RoomSettings.js:96 Original: {"name":"gg","topic":"Inception... - Events tracker: https://docs.google.com/spreadsheets/d/19c5B_khrskB_z44ETrFAh1qn0lfg0wJOEKBYohKQEEs/edit?usp=sharing","join_rule":"invite","power_levels_changed":false,"tags_changed":false,"tags":{"m.favourite":{}},"areNotifsMuted":false}
2016-03-20 08:51:30.276 RoomSettings.js:97 New: {"name":"#Marketing","join_rule":"invite","power_levels_changed":false,"tags_changed":false,"tags":{"m.favourite":{}},"areNotifsMuted":false,"_loading":false}
2016-03-20 08:51:30.278 RoomSettings.js:168 Performing 2 operations
2016-03-20 08:51:30.383 matrix.org/_matrix/client/r0/rooms/!ZzndDZQJtLEsLiMTbY%3Amatrix.org/state/m…AyMTM5NAowMDJmc2lnbmF0dXJlIINdx3xXN7zbvQRgdf1pHywhb13QckFlIlIfbwZSzQfnCg:1 PUT https://matrix.org/_matrix/client/r0/rooms/!ZzndDZQJtLEsLiMTbY%3Amatrix.org…ODAyMTM5NAowMDJmc2lnbmF0dXJlIINdx3xXN7zbvQRgdf1pHywhb13QckFlIlIfbwZSzQfnCg 400 (Bad Request)
2016-03-20 08:51:30.422 RoomView.js:824 Settings saved with 1 errors
2016-03-20 08:51:30.422 RoomView.js:827 e.e…s.MatrixError {errcode: "M_UNKNOWN", name: "M_UNKNOWN", message: "'topic' not in content", data: Object, httpStatus: 400}(anonymous function) @ RoomView.js:827(anonymous function) @ RoomView.js:826r @ q.js:834s.promiseDispatch.c @ q.js:863r.promiseDispatch @ q.js:796(anonymous function) @ q.js:604r @ q.js:137e @ q.js:125(anonymous function) @ main.js:64i.run @ browser.js:64r @ browser.js:34
2016-03-20 08:51:30.476 ScrollPanel.js:437 Scrolled to bottom; offset now 2297
```

**Comment by ara4n** [comment]
right, i see it now - thanks.

**Comment by ara4n** [comment]
fixed in matrix-org/matrix-react-sdk@31ed7b9
